src/core/lib.lua

```diff
@@ -386,12 +353,64 @@ if ffi.abi("be") then
    function htonl(b) return b end
    function htons(b) return b end
 else
-   function htonl(b) return bit.bswap(b) end
-   function htons(b) return bit.rshift(bit.bswap(b), 16) end
+   function htonl(b) return bswap(b) end
+   function htons(b) return rshift(bswap(b), 16) end
 end
 ntohl = htonl
 ntohs = htons
 
+-- The fact that BitOps return signed integers is irritating, to say
+-- the least.  One counter-intuitive consequence is that, for example,
+--
+--   bit.bswap(bit.bswap(0x80000000)) == 0x80000000
```

Comment:
@alexandergall If you think is more convenient to have that function, I think is OK. My note was more like an information note :) I'd also prefer the BitOp module returns unsigned integers, LuaJIT docs say the reason to make it signed is cross-platform compatibility.

_"Defining the result type as an unsigned number would not be cross-platform safe. All bit operations are thus defined to return results in the range of signed 32 bit numbers (converted to the Lua number type)."_

http://bitop.luajit.org/semantics.html